### ISSUE TITLE
UNKNOWN - got neither sysUptime nor sysDescr

### ISSUE THREAD

**Issue by Beatlor** [lausser/check_nwc_health#236]
Hello I see a problem with a check-result from check_nwc_health. I have a provider CPE router Cisco 3925 and authorized to grep all SNMP information out of the box and want to get out the interface datas. But obviously I can't read the sysDescr

SNMPv2-MIB::sysDescr.0 = No Such Object available on this agent at this OID
(Whyever the Cisco is here not reporting anything)

But ..
I can read the uptime
sysUpTimeInstance = Timeticks: (3609298003) 417 days, 17:49:40.03

finally the check only reports
/check_nwc_health --hostname test --community private--mode uptime
UNKNOWN - got neither sysUptime nor sysDescr, is this snmp agent working correctly?

caused by decision in modul SNMP.pm 
Line 984: if (defined $sysUptime && defined $sysDescr) {

"neither-nor" means NOT one OR the other and not AND, or I'm the wrong track?

**Comment by lausser** [comment]
Both values have to be defined. Gesendet von meinem Windows 10-Gerät. Von:  <mailto:notifications@github.com> Stephan Callsen
 Gesendet:  Freitag, 13. März 2020 08:52
 An:  <mailto:check_nwc_health@noreply.github.com> lausser/check_nwc_health
 Cc:  <mailto:subscribed@noreply.github.com> Subscribed
 Betreff:  [lausser/check_nwc_health] UNKNOWN - got neither sysUptime nor sysDescr (#236)Hello I see a problem with a check-result from check_nwc_health. I have a provider CPE router Cisco 3925 and authorized to grep all SNMP information out of the box and want to get out the interface datas. But obviously I can't read the sysDescr

SNMPv2-MIB::sysDescr.0 = No Such Object available on this agent at this OID
(Whyever the Cisco is here not reporting anything)

But ..
I can read the uptime
sysUpTimeInstance = Timeticks: (3609298003) 417 days, 17:49:40.03

finally the check only reports
/check_nwc_health --hostname test --community private--mode uptime
UNKNOWN - got neither sysUptime nor sysDescr, is this snmp agent working correctly?

caused by decision in modul SNMP.pm 
Line 984: if (defined $sysUptime && defined $sysDescr) {

"neither-nor" means NOT one OR the other and not AND, or I'm the wrong track?

-- 
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub:
https://github.com/lausser/check_nwc_health/issues/236

**Comment by Beatlor** [comment]
Thx for the answer but how can I get the informationout out of the provider Cisco? The check has no switch or option to spoof or inject the missing snmp parameter?

**Comment by Beatlor** [comment]
Now I know that the provider is actively blocking the SNMP view of sysDescr. In this case I (have to) use another check. Think you can close this issue.

**Comment by Rohlik** [comment]
I have exactly same problem with F5 BIG-IP i2800 LTM device. I also get:
`UNKNOWN - got neither sysUptime nor sysDescr, is this snmp agent working correctly?`

So that it is mean, that also F5 is not supported?

**Comment by lausser** [comment]
When you run an snmpwalk, do you get a crippled result? sysUptime and sysDescr should be among the minimum mandatory oids an snmp agent has to implement.
Does your provider/admin/... actively block sysDescr?

**Comment by Rohlik** [comment]
@lausser You are right, this was a permission issue. I did not have access to that oids. Thank you.
